### ISSUE TITLE
Cluster Architecture Question

### ISSUE THREAD

**Issue by manishiitg** [cryptagon/ion-cluster#14]
Hi @billylindeman 

i was digging deep into ion-cluster to understand how it works.
so what i understand, the way this works is 

suppose we have we have 3 ion-cluster server nodes, and all connected to a single etcd instance.

all these 3 nodes will have their own port and ip address (if they are on different server instances)

now, from the client side (js-sdk) if we connect to lets say node1 , then all peers will also get connected to peer1 only. 
if suppose 100 peers get connected to the node1 and server gets over loaded, the cluster will not automatically distribute the load to another node. all peers will keep getting connected to node1 only.

if we need to use node2, node3 we will create a different session id and connect from js-sdk to a new session. 

so basically the etcd ensure a unique session id access all the nodes, but it doesn't distribute the load across the nodes. 

So if i have use the cluster efficiently, i should many randomly connect to the different nodes i have from the client side or use some kind of metrics to decide which node to connect to.

Please do let me know if i understand this correctly or i am missing something. 

Thanks for your help!

**Comment by billylindeman** [comment]
That is correct it's meant to run behind a load balancer (ie from a cloud provider GLB/ELB, or nginx/traefik/envoy).  We use it inside kubernetes and it sites behind a K8S Service that is mapped to an Ingress.  Then all clients connect to the ingress host and kubernetes routes it to an available node

**Comment by manishiitg** [comment]
Ok, thanks understood

**Comment by manishiitg** [comment]
Hi @billylindeman 
one last question. load balancer will help to distribute load across different nodes.

but if we assume that there is a session (room) create on node1 and in that single session itself 500-1000 peers join.  and the node1 is unable to handle the load for 1000 peers so in this case the server is going to crash. 

so what i am asking is that, if there are multiple smaller rooms like 100 peer room, ion-cluster is able to server is quite well, but if on one single node there is lot of load ion-cluster is not designed to solve that problem?

Thanks again!

**Comment by billylindeman** [comment]
@manishiitg It is not designed to solve that problem (at the moment).  We will horizontally scale # of sessions, but individual sessions cannot span multiple nodes.  It is on my roadmap to solve this at some point, but not currently a priority :) 

We will probably address this in Q3 or Q4 this year
